documentation/hiera.md

```diff
@@ -0,0 +1,298 @@
+
+## Hiera configuration layers
+
+In Bolt, Hiera uses two independent layers of configuration when you look up
+data: the project layer and the module layer. This is different from Puppet,
+where Hiera has three independent layers of configuration. Bolt seraches the two
+layers of configuration in the order: project → module.
+
+### Project layer
+
+The configuration file for the project layer is located, by default, at
+`<PROJECT>/hiera.yaml`.
+
+The project layer is equivalent to the environment layer in Puppet. This is
+where you define most of your Hiera data hierarchy. Every project has a
+hierarchy configuration, which applies to all lookups made from that project.
+
+You can specify a different configuration file for the project layer by setting
+the `hiera-config` option in your project configuration, or by using the
+`--hiera-config` command-line option.
+
+📖  **Related information**
+
+- [bolt-project.yaml options](bolt_project_reference.md#hiera-config)
+- [*nix shell commands](bolt_command_reference.md)
+- [PowerShell cmdlets](bolt_cmdlet_reference.md)
+
+### Module layer
+
+The configuration for a module layer is located, by default, in the module's
+root directory at `<MODULE>/hiera.yaml`.
+
+The module layer sets default values and merge behavior for a module’s class
+parameters. The module layer comes last in Hiera’s lookup order, so environment
+data set by a user overrides the default data set by the module’s author.
+
+## Look up data from the command line
+
+If you have used Puppet, you might be familiar with the [`puppet lookup`
+command](https://puppet.com/docs/puppet/latest/hiera_automatic.html#using_puppet_lookup),
+which is the command-line interface for the `lookup()` function. Bolt does not
+have an equivalent command-line interface, so it is not possible to look up data
+with Hiera outside of running your code in Bolt.
+
+## Look up data in plans
+
+You can use the [Puppet `lookup()`
+function](https://puppet.com/docs/puppet/latest/hiera_automatic.html#puppet_lookup)
+in plans to look up data. It's useful to think of looking up Hiera data in Bolt
+plans in two different contexts: [inside apply
+blocks](applying_manifest_code.md) and outside apply blocks.
+
+### Inside apply blocks
+
+Inside apply blocks, Bolt compiles Puppet catalogs in a per-target context, and
+has unfettered access to Hiera data. You can use the same Hiera configuration
+and data you would for Puppet, and look up data in the same way as you would in
+Puppet.
+
+Before executing an apply block in a plan, you typically run the `apply_prep()`
+function on all of the targets that you run the apply on. As part of the
+`apply_prep()` function, Bolt collects facts for each target and sets them on
+the target. When Bolt compiles Puppet catalogs for each of these targets, the
+context includes the target's facts, which you can use in interpolations in your
+Hiera configuration.
+
+Given the following Hiera configuration at `<PROJECT>/hiera.yaml`:
```

Comment:
```suggestion
Given the following Hiera configuration at `<PROJECT DIRECTORY>/hiera.yaml`:
```
I know we had a PR a few months ago to standardize on this for Projects, but I'm not sure @hestonhoffman which is preferable? I don't feel strongly either way - `PROJECT` is shorter, so all things being equal I'd go that way, but we'd want to update everything else to match.